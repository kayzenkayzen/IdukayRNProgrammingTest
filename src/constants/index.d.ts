export interface IPotion {
  id: string;
  rgb: string;
  label: string;
}

export interface IAttackDamagePercentage {
  [key: number]: number;
}
