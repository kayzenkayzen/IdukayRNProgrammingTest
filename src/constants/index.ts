import { IAttackDamagePercentage, IPotion } from './index.d';

// Define the potions
const POTIONS: IPotion[] = [
  { id: 'red', rgb: '#BF2137', label: 'Poción Roja' },
  { id: 'blue', rgb: '#0479F1', label: 'Poción Azul' },
  { id: 'green', rgb: '#009E2F', label: 'Poción Verde' },
  { id: 'yellow', rgb: '#E4B80A', label: 'Poción Amarilla' },
  { id: 'gray', rgb: '#303030', label: 'Poción Gris' },
];

// Define the damage percentage
// for combinations length.
// The key is the length of the combination
// and the value is the percentage of damage
// for this combination length
const COMBINATION_DAMAGE_PERCENTAGE: IAttackDamagePercentage = {
  1: 3,
  2: 5,
  3: 10,
  4: 20,
  5: 25,
};

export const constants = {
  POTIONS,
  COMBINATION_DAMAGE_PERCENTAGE,
};
