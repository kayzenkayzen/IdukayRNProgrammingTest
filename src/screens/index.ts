export { default as PotionsSelectionScreenContainer } from '@src/screens/PotionsSelectionScreen/PotionsSelectionScreenContainer';
export { default as PotionsSelectionScreen } from '@src/screens/PotionsSelectionScreen/PotionsSelectionScreen';
export { default as OptimalDamageScreenContainer } from '@src/screens/OptimalDamageScreen/OptimalDamageScreenContainer';
export { default as OptimalDamageScreen } from '@src/screens/OptimalDamageScreen/OptimalDamageScreen';
