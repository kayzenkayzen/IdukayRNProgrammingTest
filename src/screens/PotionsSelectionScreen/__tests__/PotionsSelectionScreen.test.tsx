import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { PotionsSelectionScreen } from '@src/screens';
import { constants } from '@src/constants';

describe('Test PotionsSelectionScreen Component', () => {
  it('renders correctly', () => {
    const navigation = {};
    const removePotion = jest.fn();
    const addPotion = jest.fn();
    const calculateOptimalAttacks = jest.fn();
    const reset = jest.fn();
    const potions = constants.POTIONS.reduce(
      (accumulator, currentValue) => ({
        ...accumulator,
        [currentValue.id]: 0,
      }),
      {},
    );
    const attacks = [[]];
    const damage = 0;

    const wrapper = render(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={attacks}
        damage={damage}
      />,
    ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it("calculate button is not showing when there aren't selected potions", () => {
    const navigation = {};
    const removePotion = jest.fn();
    const addPotion = jest.fn();
    const calculateOptimalAttacks = jest.fn();
    const reset = jest.fn();
    const potions = constants.POTIONS.reduce(
      (accumulator, currentValue) => ({
        ...accumulator,
        [currentValue.id]: 0,
      }),
      {},
    );
    const attacks = [[]];
    const damage = 0;

    const { queryByTestId } = render(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={attacks}
        damage={damage}
      />,
    );

    expect(queryByTestId('calculate-optimal-attacks-button')).toBeNull();
  });

  it('calculate button is showing when there is one or more selected potions', () => {
    const navigation = {};
    const removePotion = jest.fn();
    const addPotion = jest.fn();
    const calculateOptimalAttacks = jest.fn();
    const reset = jest.fn();
    const potions = constants.POTIONS.reduce(
      (accumulator, currentValue) => ({
        ...accumulator,
        [currentValue.id]: 1,
      }),
      {},
    );
    const attacks = [[]];
    const damage = 0;

    const { queryByTestId } = render(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={attacks}
        damage={damage}
      />,
    );

    expect(queryByTestId('calculate-optimal-attacks-button')).toBeDefined();
  });

  it('calculateOptimalAttacks callback prop is called', () => {
    const navigation = {};
    const removePotion = jest.fn();
    const addPotion = jest.fn();
    const calculateOptimalAttacks = jest.fn();
    const reset = jest.fn();
    const potions = constants.POTIONS.reduce(
      (accumulator, currentValue) => ({
        ...accumulator,
        [currentValue.id]: 1,
      }),
      {},
    );
    const attacks = [[]];
    const damage = 0;

    const { getByTestId } = render(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={attacks}
        damage={damage}
      />,
    );

    fireEvent.press(getByTestId('calculate-optimal-attacks-button'));

    expect(calculateOptimalAttacks).toHaveBeenCalledTimes(1);
  });

  it('attacks and damage props update call useEffect correctly', () => {
    const navigation = {
      navigate: jest.fn(),
    };
    const removePotion = jest.fn();
    const addPotion = jest.fn();
    const calculateOptimalAttacks = jest.fn();
    const reset = jest.fn();
    const potions = constants.POTIONS.reduce(
      (accumulator, currentValue) => ({
        ...accumulator,
        [currentValue.id]: 1,
      }),
      {},
    );
    const attacks = [[]];
    const damage = 0;

    const { rerender } = render(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={attacks}
        damage={damage}
      />,
    );

    rerender(
      <PotionsSelectionScreen
        navigation={navigation}
        removePotion={removePotion}
        addPotion={addPotion}
        calculateOptimalAttacks={calculateOptimalAttacks}
        reset={reset}
        potions={potions}
        attacks={[['red', 'blue'], ['red']]}
        damage={10}
      />,
    );

    expect(navigation.navigate).toHaveBeenCalledTimes(1);
    expect(reset).toHaveBeenCalledTimes(1);
  });
});
