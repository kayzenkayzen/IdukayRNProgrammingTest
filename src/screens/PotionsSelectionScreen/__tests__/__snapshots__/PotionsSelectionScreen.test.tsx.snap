// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PotionsSelectionScreen Component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "paddingHorizontal": 10,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2b2b2b",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 16,
              "lineHeight": 21,
            }
          }
        >
          Seleccione las pociones que posee para calcular los ataques más efectivos
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#dddddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginVertical": 2,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            },
            Object {
              "borderTopColor": "#ddd",
              "borderTopWidth": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/potion.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#BF2137",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#BF2137",
                "fontFamily": "SourceSansPro-SemiBold",
              }
            }
            testID="potion-option-item-label"
          >
            Poción Roja
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopLeftRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="minus-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b30",
                    },
                  ]
                }
              >
                -
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#000",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 60,
                  },
                  Object {
                    "borderColor": "#c0c0c0",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                  ]
                }
                testID="value-label"
              >
                0
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopEndRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="more-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b",
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#dddddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginVertical": 2,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            },
            Object {
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/potion.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#0479F1",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#0479F1",
                "fontFamily": "SourceSansPro-SemiBold",
              }
            }
            testID="potion-option-item-label"
          >
            Poción Azul
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopLeftRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="minus-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b30",
                    },
                  ]
                }
              >
                -
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#000",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 60,
                  },
                  Object {
                    "borderColor": "#c0c0c0",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                  ]
                }
                testID="value-label"
              >
                0
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopEndRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="more-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b",
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#dddddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginVertical": 2,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            },
            Object {
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/potion.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#009E2F",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#009E2F",
                "fontFamily": "SourceSansPro-SemiBold",
              }
            }
            testID="potion-option-item-label"
          >
            Poción Verde
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopLeftRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="minus-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b30",
                    },
                  ]
                }
              >
                -
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#000",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 60,
                  },
                  Object {
                    "borderColor": "#c0c0c0",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                  ]
                }
                testID="value-label"
              >
                0
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopEndRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="more-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b",
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#dddddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginVertical": 2,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            },
            Object {
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/potion.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#E4B80A",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#E4B80A",
                "fontFamily": "SourceSansPro-SemiBold",
              }
            }
            testID="potion-option-item-label"
          >
            Poción Amarilla
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopLeftRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="minus-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b30",
                    },
                  ]
                }
              >
                -
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#000",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 60,
                  },
                  Object {
                    "borderColor": "#c0c0c0",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                  ]
                }
                testID="value-label"
              >
                0
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopEndRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="more-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b",
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomColor": "#dddddd",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "marginVertical": 2,
              "paddingHorizontal": 10,
              "paddingVertical": 10,
            },
            Object {
              "borderTopColor": undefined,
              "borderTopWidth": undefined,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/potion.png",
              }
            }
            style={
              Array [
                Object {
                  "height": 20,
                  "marginRight": 10,
                  "width": 20,
                },
                Object {
                  "tintColor": "#303030",
                },
              ]
            }
          />
          <Text
            style={
              Object {
                "color": "#303030",
                "fontFamily": "SourceSansPro-SemiBold",
              }
            }
            testID="potion-option-item-label"
          >
            Poción Gris
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomLeftRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopLeftRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="minus-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b30",
                    },
                  ]
                }
              >
                -
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "borderColor": "#000",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                    "marginHorizontal": 5,
                    "width": 60,
                  },
                  Object {
                    "borderColor": "#c0c0c0",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                  ]
                }
                testID="value-label"
              >
                0
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomEndRadius": 20,
                  "borderColor": "#c0c0c0",
                  "borderRadius": 4,
                  "borderTopEndRadius": 20,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "opacity": 1,
                  "width": 40,
                }
              }
              testID="more-button"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "SourceSansPro-Bold",
                      "fontSize": 16,
                    },
                    Object {
                      "color": "#2b2b2b",
                    },
                  ]
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
